### ISSUE TITLE
HADOOP_CONF_DIR is ignored when it already exists

### ISSUE THREAD

**Issue by kiwy42** [gupta-paras/pyhdfs_client#1]
Hello,  
Most installation of hadoop do not store configuration inside `$HADOOP_HOME/etc`  
Is it possible to use `$HADOOP_CONF_DIR` instead of forging it is it already is set ?  
I'm not very good with python and git so I'm not really able to submit a patch.but it should like like this :   

```
def update_hadoop_environ(self):
        if not `HADOOP_CONF_DIR` in os.environ:
            os.environ['HADOOP_CONF_DIR'] = os.path.join(os.environ['HADOOP_HOME'], r'etc\hadoop')
        if not `YARN_CONF_DIR` in os.environ:
            os.environ['YARN_CONF_DIR'] = os.environ['HADOOP_CONF_DIR']
        if not `HADOOP_PREFIX` in os.environ:
            os.environ['HADOOP_PREFIX'] = os.environ['HADOOP_HOME']
        os.environ['PATH'] = os.environ['PATH'] + ";" + os.path.join(os.environ['HADOOP_HOME'], r'bin')
        if not sys.platform.lower().startswith('win'):
            os.environ['PATH'] = os.environ['PATH'].replace(";", ":").replace("\\", '/')
            os.environ['HADOOP_CONF_DIR'] = os.environ['HADOOP_CONF_DIR'].replace(";", ":").replace("\\", '/')
            os.environ['YARN_CONF_DIR'] = os.environ['YARN_CONF_DIR'].replace(";", ":").replace("\\", '/')
```

Instead of creating variable it should be better to use the one already available.

**Comment by kiwy42** [comment]
Hum, though it was a good idea but it in fact doesn't work.
I don"t know why but it seems the way hadoop is loaded in py4j do not account for `$HADOOP_CONF_DIR`

**Comment by gupta-paras** [comment]
Hi @kiwy42 
Thanks a lot for identifying above. 
I will try to use system set environment variables and default only if they won't exist.
Also, you can submit the above patch, I will test and add it to the code.

**Comment by kiwy42** [comment]
I'm not really aware on how to propose patch on github and in code revisionning.
However though working I cannot manage to connect to an HA hdfs cluster, I've tried several thing : 

```
from pyhdfs_client.pyhdfs_client import HDFSClient
hdfs_client = HDFSClient()
ret, out, err = hdfs_client.run(['-ls', '/'])
ret, out, err = hdfs_client.run(['-ls', 'hdfs://clustername/'])
```
the first one using `/` I just list my local root folder, but by using 'hdfs://clustername/' it doesn"t work,   
I have been crawling in a lot about hadoop configuration java objet, it seems that hdfs configuration is not loaded by the object `org.apache.hadoop.conf.Configuration`  and all java code I've found set hdfs configuration manually, I may misunderstand something but I though as it's using hadoop code would automatically load configuration from `hdfs-site.xml`.   
I'm not fluent in Python and it's possible I do not understand how you're using py4j.  
My current configuration works from cli with command line such as `hdfs dfs -ls /` without issue. I'm missing something I'm sure. maybe it should be another issue open.
